### PR TITLE
Update gradle to 8.8 and AGP to 8.5.0

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
Update gradle to 8.8 and AGP to 8.5.0

**Alternative(s) considered**
NA

**Type**
Builds

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (aditya-07) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can we use the latest gradle version available?

**Review by aditya-07 — COMMENTED** [review]
Check to see if additional changes are required.
Also, is there a reason if we are updating to these specific versions?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jingtang10) [reply gradle/wrapper/gradle-wrapper.properties]: done

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -11,7 +11,7 @@ repositories {
 dependencies {
   implementation("com.diffplug.spotless:spotless-plugin-gradle:6.22.0")
 
-  implementation("com.android.tools.build:gradle:8.1.4")
+  implementation("com.android.tools.build:gradle:8.4.1")
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply buildSrc/build.gradle.kts]: done

**Comment by jingtang10** [comment]
addressed comments
thanks aditya for reviewing

**Review by MJ1998 — COMMENTED** [review]
Have you verified compatibility of other dependencies' with this agp? 
I guess our test suite should be enough ?

Update: Some tests failed which is a good sign!

**Comment by jingtang10** [comment]
After taking a look at the failure:

```
Lint found 2 errors, 38 warnings. First failure:
[1185](https://github.com/google/android-fhir/actions/runs/9566698928/job/26372611822?pr=2575#step:5:1186)
[1186](https://github.com/google/android-fhir/actions/runs/9566698928/job/26372611822?pr=2575#step:5:1187)/home/runner/work/android-fhir/android-fhir/datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreLocalDates.kt:128: Error: Implicit cast from IsoChronology to Chronology requires API level 26 (current min is 24) [NewApi]
[1187](https://github.com/google/android-fhir/actions/runs/9566698928/job/26372611822?pr=2575#step:5:1188)    IsoChronology.INSTANCE,
[1188](https://github.com/google/android-fhir/actions/runs/9566698928/job/26372611822?pr=2575#step:5:1189)    ~~~~~~~~~~~~~~~~~~~~~~
```

and trying to fix it without updating min sdk version, I realised we do need the ISOChronology class in order to get the localized date time pattern... without using this class, it's possible to generate the date time formatter object that is localized, but we won't be able to get the actually localized pattern e.g. yyyy-mm-dd (by calling the api `getLocalizedDateTimePattern`)... but we do need this pattern itself as the hint text...

now an alternative is to not use that as hint text, but instead use a formatted date e.g. 2019-09-12... which might be ok but it will just diverge a bit from the behaviour we have when the entry format string is provided by the questionnaire.

so i think the easiest thing to do is actually update the minsdk.. which i think wasn't an issue only because of the build tool wasn't picking it up since the api was indeed introduced in api level 26... so this way we're only making what was already the case explicit... (device on 24 would have had problem with this already).

**Comment by MJ1998** [comment]
We dropped the minSdk (for all libraries) to 24 due to this issue #2077.

**Comment by pld** [comment]
Just to confirm, we're fine raising the min level to 26

**Comment by jingtang10** [comment]
discussed today at the developers call and no objection so far to this api level change.

if we don't hear anything else we'll include this change in the next release

**Review by aditya-07 — APPROVED** [review]
LGTM.

**Pull request merged**